Reports tab

vishk1 opened the pull request:
Issue #434 
- Created Reports.vue
- Created Components/Reports folder to store components
- Created all 13 components
           - ActiveTeammates.vue
           - AvgWaitTime.vue
           - ClaimedWorksites.vue
           - CompletedWorksites.vue
           - EstCommercialValue.vue
           - ActiveTeammates.vue
           - InactiveTeammates.vue
           - IncompleteWorksites.vue
           - LongestWaitTime.vue
           - ShortestWaitTime.vue
           - TeammatesInvited.vue
           - UnassignedWorksites.vue
           - VolunteerWorkLogs.vue
           - WorksiteCompletion.vue
- within each component:
           - used flex-grids to position items
           - added temp star & lock icons
           - added content
- Reports.vue
           - imported all 13 components
           - used flex-grids to position each component
           - used same-sized cards to display information from each component

- Navigation
           - integrated to router.js
           - made "authenticated"
           - added to side navigation bar in "authenticated.vue"

vishk1 commented:
![image](https://user-images.githubusercontent.com/60664816/77737852-c661ae00-6fdc-11ea-9495-9e2767b9341a.png)

vishk1 commented:
![image](https://user-images.githubusercontent.com/60664816/77738101-3839f780-6fdd-11ea-9ee6-168b133a018b.png)
![image](https://user-images.githubusercontent.com/60664816/77738129-46881380-6fdd-11ea-9f0c-e767f9f70fbb.png)

The scroll screenshots were super choppy, so these are more clear pictures

vishk1 commented:
Issue #437 
- Changed Headers
- Reworded Mini-Cards to fit new titles/descriptions
- imported new cards
         - EstCommercialValue.vue
         - OrgParticipationStatus
         - TeammateInvitation.vue
         - IncidentStatusReport
         - CallCenterFlow.vue
         - RequestsCard.vue
- destroyed old cards
         - avg/shortest/longest time cards
         - TeammatesInvited.vue
- re-positioned cards
         - download csv (with blank link) under "included" section
         - favorites empty
         - everything else underneath "Reports Library"
- commented out stars
- resized cards to properly fit content
- Created "ActivationPopUp.vue" modal

---- still working on -----
- showing modal on click of card under "reports library"
- sending report of whoever clicks to request access (regardless if they email or not)

BradenM commented:
You should be able to find the SVG for the nav icon in zeplin. Let me know if you can't. 

Also, remember to add WIP if it's not ready for merge 😉

vishk1 commented:
**Updated Reports Tab**
![image](https://user-images.githubusercontent.com/60664816/77860411-b9a7aa80-71d4-11ea-8551-b1a1c30b0fea.png)
![image](https://user-images.githubusercontent.com/60664816/77860438-e8258580-71d4-11ea-9a4e-dabdfafeeed2.png)

BradenM commented on src/layouts/Authenticated.vue:
```diff
@@ -165,6 +165,12 @@ export default {
+        {
+          key: 'reports',
+          icon: 'history',
+          text: this.$t('dashboard.reports'),
```
If you remove the `text` element here, the `NavButton` component will use the `key` item and fill in the correct localization

erics2783 commented:
@BradenM @vishk1 I rebased this branch on to master since it was a good bit behind. I also added the icon to the reports link and disabled it if the user is not in a government organization per issue #446. This requires changes in https://github.com/CrisisCleanup/crisiscleanup-3-api/pull/162 to be merged.

BradenM commented:
@erics2783 @vishk1 https://github.com/CrisisCleanup/crisiscleanup-3-api/pull/162 merged.

vishk1 commented:
Issue #437 
---- activation pop up ----
- now upon click of any card under "reports library" the activation pop up will appear
- fixed modal syntax
- changed prop to 'isShowingModal' on Reports.vue
- changed from using data to prop to trigger action in Reports.vue
- deleted "show Modal" button from ActivationPopUp.vue
- tried numerous methods to combat this ongoing issue
- created modal within Reports.vue (similar to Affiliates.vue)
- changed modal prop from "isShowingModal" to "showRequestAccessModal"
- added button action as trigger for modal
- wrapped each card in buttons to allow for clickability
---- misc ----
- fixed text-centering issue once buttons were introduced on cards
- fixed numerous syntax issues
- fixed card sizing
- fixed card placement
- fixed card spacing (adjusted side and up/down margins separating them)
- fixed content placement issue (got rid of the forehead on some cards)

vishk1 commented:
**Resized Cards**
![image](https://user-images.githubusercontent.com/60664816/78468968-2abdf500-76e2-11ea-997a-51899525f0cf.png)
**Pop Up**
![image](https://user-images.githubusercontent.com/60664816/78468988-4d500e00-76e2-11ea-873b-f7b0b73f3f99.png)

tabiodun commented:
@BradenM @vishk1 I've fixed this up and rebased into one commit so that I can merge it.
